app/models/product.rb

```diff
@@ -763,21 +775,36 @@ def variables(_options = {})
     list
   end
 
+  def net_surface_area
+    computed_surface = self.reading_cache[:net_surface_area] || self.reading_cache['net_surface_area']
+    return computed_surface if computed_surface
+    self.net_surface_area = calculation_of_net_surface_area
+  end
+
   # Override net_surface_area indicator to compute it from shape if
   # product has shape indicator unless options :strict is given
-  def net_surface_area(options = {})
+  def calculation_of_net_surface_area(options = {})
```

Comment:
Please consider to rename method to `calculate_net_surface_area`. An action verb at start is better.